app/src/main/java/com/fbada/projectone/Car.kt

```diff
@@ -0,0 +1,349 @@
+    /**
+     * @property motCheck checks for valid MOT
+     */
+    val motCheck: Boolean = true
+
+    /**
+     * @property doorPremium multiplies number of doors by 1.5
+     */
+    val doorPremium: Double
+        get() = doors * 1.5  // Getter
+    /**
+     * @property price price of the car based on doors property.
+     *  If a car has less than 5 doors 10 is added to base price.
+     *  If car has 5 to 8 doors 20 is added to base price, if there are more than 8 doors 33 is added.
+     */
+    override var price: Double = 0.00 + if (doors >= 5) {
+            if (doors > 8) {
+                super.price + 33.00
+            } else super.price + 20.00
+        } else {
+            super.price + 10.00
+        }
+}
+/**
+ * Subclass Ford Manufacturer type car.  Extends Car class.
+ */
+open class Ford : Car(2020, AvailableColours.Red, 300.00) {
+    /**
+     * @property seats number of seats
+     * @property location pickup location
+     */
+    var seats: Int = 8
+    val location: String = "Manchester"
+    /**
+     * @property price base price becomes 100.00
+     */
+    override var price: Double = 100.00
+
+    /**
+     * Start test drive
+     */
+    fun testDrive() {
+        println("Lets take this for a spin!")
+    }
+
+    /**
+     * Opens the boot
+     */
+    open fun openBoot() {
+        println("boot is open")
+    }
+
+
+}
+
+/**
+ * Subclass Alpha Model type car.  Extends BMW class.
+ * @property dashboard Big or small dashboard type
+ * @property satNav Sat Nav available
+ * @property ecoRating Eco Rating Score
+ */
+class Alpha(private val dashboard: String, val satNav: Boolean, val ecoRating: Int) : BMW("Diesel", "Manual") {
+    /**
+     * @property price Base price plus eco rating multiplied by 2.
+     */
+    override var price: Double = super.price + (ecoRating * 2)
+    /**
+     * Navigation function which prints a message dependent on whether the dashboard is big or small.
+     */
+    override fun navigate() {
+        if (dashboard == "big") {
+            println("Alpha is so easy to navigate")
+        } else {
+            println("Alpha map is too small")
+        }
+    }
+}
+
+/**
+ * Subclass Beta Model type car.  Extends BMW class.
+ * @property insurance Insurance available
+ * @property dashboard Big or small dashboard type
+ * @property powerSteering Power steering available
+ */
+class Beta(private val insurance: Boolean, private val dashboard: String, private val powerSteering: Boolean) :
+    BMW("Petrol", "Manual") {
+    /**
+     * Navigation function which prints a message dependent on whether the dashboard is big or small.
+     */
+    override fun navigate() {
+        if (insurance) {
+            if (dashboard == "big") {
+                price * 40
+                println("Beta is so easy to navigate")
+            } else {
+                price / 3
+                println("Beta map is too small")
```

Comment:
Those price operations don't result in anything. Price isn't reassigned, that's literally an "empty" execution, nothing happens.
On a side note - even if the the result of execution would have been assigned to the `price` prop, why would we need to change the price while navigating?

Reply:
Code block doesn't make logical sense.  Refactored to make more sense.  May further refactor with hierarchy refactor.